lib/tasks/one_off.rake

```diff
@@ -279,17 +279,34 @@ namespace :one_off do
     processed_count = 0
 
     puts "Processing...."
-    affected_projects.in_batches(of: batch_size).each do |batch|
-      batch.each do |project|
-        affected_versions = project.versions.where(%((repository_sources != '["Maven"]' AND repository_sources != '["Google"]') OR repository_sources IS NULL))
-
-        puts "Destroying #{affected_versions.count} versions for #{project.platform}/#{project.name}."
-        affected_versions.destroy_all if commit
+    affected_projects.in_batches(of: batch_size).each do |projects_batch|
+      projects_batch.each do |project|
+        no_source_versions = project.versions.where("repository_sources IS NULL")
+        ignored_source_versions = project.versions.where(%((repository_sources != '["Maven"]' AND repository_sources != '["Google"]')))
+
+        puts "Updating/Deleting #{no_source_versions.count + ignored_source_versions.count} versions for #{project.platform}/#{project.name}."
+        if commit
+          # If no repository_sources, then destroy it
+          no_source_versions.destroy_all
+
+          # If there are repository_sources,
+          # - if any are Maven or Google, remove repository_sources that aren't Maven or Google
+          # - if none are Maven or Google, destroy version
+          ignored_source_versions.in_batches do |versions_batch|
+            versions_batch.each do |version|
+              if version.repository_sources.include?("Maven") || version.repository_sources.include?("Google")
```

Comment:
Do the repository_sources use a consistent casing or are there variations of 'maven' that we'd want to keep?

Reply:
They should be consistent, given my understanding!